phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java

```diff
@@ -0,0 +1,231 @@
+            // run the index MR job.
+            IndexToolIT.runIndexTool(true, false, null, dataTableName, indexName);
+        }
+        // Configure Indexer to skip the last write phase (i.e., the post index update phase) where the verify flag is set
+        // to true and/or index rows are deleted and check that this does not impact the correctness
+        Indexer.setSkipPostIndexUpdatesForTesting(true);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
+        conn.commit();
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val1, val2) values ('c', 'cd','cde')");
+        conn.commit();
+        String selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        ResultSet rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("abcc", rs.getString(1));
+        assertEquals("abcd", rs.getString(2));
+        assertFalse(rs.next());
+        conn.close();
+    }
+
+    @Test
+    public void skipDataTableAndPostIndexPartialRowUpdate() throws Exception {
+        String dataTableName = generateUniqueName();
+        populateTable(dataTableName);
+        Connection conn = DriverManager.getConnection(getUrl());
+        String indexName = generateUniqueName();
+        conn.createStatement().execute("CREATE INDEX " + indexName + " on " +
+                dataTableName + " (val1) include (val2, val3)" + (async ? "ASYNC" : ""));
+        if (async) {
+            // run the index MR job.
+            IndexToolIT.runIndexTool(true, false, null, dataTableName, indexName);
+        }
+        // Configure Indexer to skip the last two write phase (i.e., the data table update and post index update phase)
+        // and check that this does not impact the correctness
+        Indexer.setSkipDataTableUpdatesForTesting(true);
+        Indexer.setSkipPostIndexUpdatesForTesting(true);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
+        conn.commit();
+        Indexer.setSkipDataTableUpdatesForTesting(false);
+        Indexer.setSkipPostIndexUpdatesForTesting(false);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val3) values ('a', 'abcdd')");
+        conn.commit();
+        String selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        ResultSet rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("abc", rs.getString(1));
+        assertEquals("abcdd", rs.getString(2));
+        assertFalse(rs.next());
+        conn.close();
+    }
+}
```

Comment:
Recommend to add tests for 2 indexes on the same table that has the same column. If the data table is updated, both of them should be valid.

Reply:
I will add a test to for testing multiple indexes on a data table.

Reply:
I changed one of the tests to use two indexes.